### PR TITLE
Format currencies improvements

### PR DESCRIPTION
**Pull request by ptbrowne**
When we detect a currency in the data (from the unit in qudt.unit), we format the data with currencyExponent digits.

In this issue, the usecase of data where the floating precision is more than currencyExponent (for example for currency exchange datasets) is not tackled. It would be possible to tackle this when the dimension metadata informs us on the precision necessary.

Fixes https://github.com/orgs/visualize-admin/projects/1/views/18

This can be tested with the "State accounts - Office" dataset where amounts should have a precision of 2 digits after comma since the currency is CHF, and the currency exponent of CHF is 2. 

image

### CODE REVIEW

File: app/configurator/components/ui-helpers.spec.ts [hunk 1]
```diff
@@ -75,6 +83,11 @@ describe("useDimensionFormatters", () => {
     const { formatters } = setup();
     expect(formatters["iri-number"]("2.33333")).toEqual("2,33");
   });
+
+  it("should work with currencies", () => {
+    const { formatters } = setup();
+    expect(formatters["iri-currency"]("20002.3333")).toEqual("20'002,3");
+  });
```
Comment (bprusinowski) [inline app/configurator/components/ui-helpers.spec.ts:90]: Just wondering, if you're using locale in this formatter (currencyFormatter), can the test fail in some scenarios if run with a locale that use different formatting? Or will it be the same in all cases?

**Review by bprusinowski — APPROVED** [review]
LGTM 👌

File: app/rdf/query-hierarchies.ts [hunk 2]
```diff
@@ -59,6 +60,29 @@ const toTree = (
...
 };
 
+const findHierarchyForDimension = (cube: Cube, dimensionIri: string) => {
+  const newHierarchy = cube.ptr
+    .any()
+    .has(ns.sh.path, rdf.namedNode(dimensionIri))
+    .has(ns.cubeMeta.inHierarchy)
+    .out(ns.cubeMeta.inHierarchy)
+    .toArray()
+    .shift();
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/rdf/query-hierarchies.ts:70]: To get the first element, it's clearer with [0] I updated the code.

File: app/configurator/components/ui-helpers.spec.ts (hunk 1 above)
Comment (ptbrowne) [reply app/configurator/components/ui-helpers.spec.ts:90]: It could fail if run with a different locale, but for now we do not test for this.

**Pull request merged**